eos-paygd/main.c

```diff
@@ -35,8 +35,90 @@
 #include <sys/un.h>
 #include <libeos-payg/efi.h>
 
+#define LOGFILE_DIRNAME "/var/log/eos-payg"
+#define LOGFILE_BASENAME "eos-paygd"
+#define LOGFILE_EXT "log"
+
 #define TIMEOUT_POWEROFF_ON_ERROR_MINUTES 20
 
+static GIOChannel *
+open_log_file (void)
+{
+  const gchar *log_file_name = NULL;
+  g_autoptr(GFile) log_file = NULL;
+  g_autoptr(GDateTime) now = NULL;
+  g_autofree gchar *tstamp = NULL;
+
+  /* Build log file name with a date stamp so we get one file per day,
+   * ex., eos-paygd-20220418.log
+   */
+  now = g_date_time_new_now_local ();
+  tstamp = g_date_time_format (now, "%Y%m%d");
+  log_file_name = g_strconcat (LOGFILE_DIRNAME, "/", LOGFILE_BASENAME, "-",
+                               tstamp, ".", LOGFILE_EXT, NULL);
+  log_file = g_file_new_for_path (log_file_name);
+
+  /* We can't log an error here if this fails, as it would cause infinite
+   * recursion */
+  return g_io_channel_new_file (log_file_name, "a", NULL);
+}
+
+/* we can't use g_log_writer_default_would_drop until glib 2.68, which will be
```

Comment:
Nitpick: I would have used (pseudocode) `#if GLIB_CHECK_VERSION(2, 68, 0) { return g_log_writer_default_would_drop (…) } #else { … simplified fallback … }` so that it’s easier to find and drop the fallback code in future (by grepping for `GLIB_CHECK_VERSION`). Again, just an fyi, not worth changing now.